language/stdlib/modules/validator_config.move

```diff
@@ -1,156 +1,156 @@
-        Transaction::assert(LibraAccount::exists(delegated_account), 5);
-        // check delegated address is different from transaction's sender
-        Transaction::assert(delegated_account != Transaction::sender(), 6);
-        let t_ref = borrow_global_mut<T>(Transaction::sender());
-        t_ref.delegated_account = Option::some(delegated_account)
-    }
-
-    // If the sender of the transaction has a ValidatorConfig::T resource,
-    // this function will revoke management capabilities from the delegated_account account.
-    public fun remove_delegated_account() acquires T {
-        let t_ref = borrow_global_mut<T>(Transaction::sender());
-        t_ref.delegated_account = Option::none()
+        full_node_network_address: vector<u8>,
+    ) acquires T {
+        Transaction::assert(Transaction::sender() ==
+                            get_operator(validator_account), 1101);
+        // TODO(valerini): verify the validity of new_config.consensus_pubkey and
+        // the proof of posession
+        let t_ref = borrow_global_mut<T>(validator_account);
+        t_ref.config = Option::some(Config {
+            consensus_pubkey,
+            validator_network_signing_pubkey,
+            validator_network_identity_pubkey,
+            validator_network_address,
+            full_node_network_identity_pubkey,
+            full_node_network_address,
+        });
     }
 
-    // Rotate validator's config.
-    // Here validator_account - is the account of the validator whose
-    // consensus_pubkey is going to be rotated.
+    // TODO(valerini): to remove and call into rotate_config instead
     public fun rotate_consensus_pubkey(
         validator_account: address,
-        new_consensus_pubkey: vector<u8>,
-        // _proof: vector<u8>
+        consensus_pubkey: vector<u8>,
     ) acquires T {
-        let addr = get_validator_operator_account(validator_account);
-        Transaction::assert(Transaction::sender() == addr, 1);
+        Transaction::assert(Transaction::sender() ==
+                            get_operator(validator_account), 1101);
+        let t_config_ref = Option::borrow_mut(&mut borrow_global_mut<T>(validator_account).config);
+        t_config_ref.consensus_pubkey = consensus_pubkey;
+    }
 
-        // TODO(valerini): verify the proof of posession of new_consensus_secretkey
+    ///////////////////////////////////////////////////////////////////////////
+    // Publicly callable APIs: getters
+    ///////////////////////////////////////////////////////////////////////////
 
-        let t_ref = borrow_global_mut<T>(validator_account);
-        // Set the new key
-        t_ref.config.consensus_pubkey = new_consensus_pubkey;
+    // Returns true if all of the following is true:
+    // 1) there is a ValidatorConfig resource under the address, and
+    // 2) the config is set, and
+    // NB! currently we do not require the the operator_account to be set
+    public fun is_valid(addr: address): bool acquires T {
```

Comment:
and the validator operator is valid

Reply:
Right, I will add the ValidatorOperator role in the next PR (may be by then the role model will get settled on).